### ISSUE TITLE
Http user agent changed after 3.6.1 release

### ISSUE THREAD

**Issue by TomasValenta** [bumptech/glide#546]
Hello,
in 3.6.0 release the cusotm Http user agent look like [this](https://cloud.githubusercontent.com/assets/4901251/8768337/7afe8cc2-2e7c-11e5-8fb5-577e10c48dd7.png):
```
HTTP_USER_AGENT : Mozilla/5.0 (Windows NT 6.3; WOW64; rv:37.0) Gecko/20100101 Firefox/37.0
```
after 3.6.1 release the same cusotm Http user agent look like [this](https://cloud.githubusercontent.com/assets/4901251/8768340/8fd0ce62-2e7c-11e5-8647-65afe572fd6a.png):
```
HTTP_USER_AGENT : Dalvik/2.1.0 (Linux; U; Android 5.1.1; LG-D855 Build/LMY48G),Mozilla/5.0 (Windows NT 6.3; WOW64; rv:37.0) Gecko/20100101 Firefox/37.0
```
It's a bug or feature? Because in my app this change brought new problems ;)

**Comment by TWiStErRob** [comment]
Hmm, maybe both...
How do you set your custom user agent?
Do you use OkHttp integration or just Glide?

There has been some changes to user agent handling, see [3.6.1 release notes](https://github.com/bumptech/glide/releases/tag/v3.6.1) and https://github.com/bumptech/glide/compare/v3.6.0...v3.6.1.
As far as I can see every precaution has been made ([`isUserAgentDefault`](https://github.com/bumptech/glide/compare/v3.6.0...v3.6.1#diff-0d69682540c80f964a21f69e247fbb99R146)) to prevent concatenating two `User-Agent` headers, which is what you have if I see it correctly.

**Comment by TomasValenta** [comment]
Hello Róbert,

I use just Glide, the new feature from 3.6.0 release. My usage:

```java
Glide.with(context)
	.load(HttpHeader.getUrl(source))
	.into(ViewHolder.Image);
```

HttpHeader class:

```java
public class HttpHeader {

    public static final String USER_AGENT = "Mozilla/5.0 (Windows NT 6.3; WOW64; rv:37.0) Gecko/20100101 Firefox/37.0";
    public static final String APP_AGENT = "XYZ";

    public static GlideUrl getUrl(String url) {
        if (TextUtils.isEmpty(url)) {
            return null;
        }
        else return new GlideUrl(url, new LazyHeaders.Builder()
                .addHeader("User-Agent", USER_AGENT)
                .addHeader("App-agent", APP_AGENT)
                .build());
    }
}
```

**Comment by TWiStErRob** [comment]
@sjudd confirmed, it's a bug, it looks like the `isUserAgentDefault` is always initialized false: https://github.com/bumptech/glide/commit/487d9c5ea488470fc3796e582467baf1c5078f7a#diff-0d69682540c80f964a21f69e247fbb99R120

```java
LazyHeaders headers = new Builder()
		.addHeader("User-Agent", "My user agent")
		.addHeader("App-agent", "My app agent")
		.build();

System.out.println(headers);
System.out.println(headers.getHeaders());
```

3.6.0:
```
LazyHeaders{headers={App-agent=[StringHeaderFactory{value='My app agent'}], User-Agent=[StringHeaderFactory{value='My user agent'}]}}
{App-agent=My app agent, User-Agent=My user agent}
```

3.6.1:
```
LazyHeaders{headers={App-agent=[StringHeaderFactory{value='My app agent'}], Accept-Encoding=[StringHeaderFactory{value='identity'}], User-Agent=[StringHeaderFactory{value='Dalvik/1.6.0 (Linux; U; Android 4.4.2; GT-I9505 Build/KOT49H)'}, StringHeaderFactory{value='My user agent'}]}}
{Accept-Encoding=identity, App-agent=My app agent, User-Agent=Dalvik/1.6.0 (Linux; U; Android 4.4.2; GT-I9505 Build/KOT49H),My user agent}
```

@TomasValenta workaround is really simple if you want to use Glide 3.6.1 (`addHeader` -> `setHeader`):
```java
LazyHeaders headers = new Builder()
		.setHeader("User-Agent", "My user agent")
		.addHeader("App-agent", "My app agent")
		.build();
```

**Comment by TomasValenta** [comment]
Thank you Róbert!

**Comment by sjudd** [comment]
Urk, thanks for reporting this, I'll take look!

**Comment by TWiStErRob** [comment]
@sjudd This looks like it's been fixed on your 3.0 branch and has been also merged to bumptech/glide 3.0 branch, but not yet released (scheduled for 3.6.2 or 3.7.0). I thought merging a "fixes #???" commit will close the issue, but it seems that only works on master or the default branch. Closing this manually because I see the commit on the right branch.

**Comment by TomasValenta** [comment]
Fixed. Thank you!
